src/objectExplorer/objectExplorerService.ts

```diff
@@ -277,15 +308,19 @@ export class ObjectExplorerService {
                         let node = await promise;
                         // if password failed
                         if (!node) {
-                            const connectNode = new ConnectTreeNode(element);
-                            this._treeNodeToChildrenMap.set(element, [connectNode]);
-                            return [connectNode];
+                            // remove password if it's saved in the credential store
```

Comment:
How do you know that the failure was because of a wrong password here? What happens if something else is wrong - such as a user doesn't have connect permissions, wrong username, etc?

Reply:
Changed the comment

Reply:
Ok, but why are you removing the password in all cases? What if the connection fails because of a network failure for example? I wouldn't expect (or want) the stored password to be removed in that case.

Reply:
Because if a connection wasn't made, there's no need to save password. We should only save password if a connection was made atleast once (a profile was made).

Reply:
What about an existing connection that I'm reconnecting? So I successfully make and save a connection and then later on disconnect and try to reconnect but some general network error occurs. Won't it strip the saved password at that point so if I try to connect again it'll prompt me for the password even though the error had nothing to do with a wrong password?

Reply:
Ok, so looking through it more closely it'll remove it from the profile as long as it was created and stored in the credential store. That seems fine then. Although I'm curious why you need to check if it's saved in the credential store - would you ever expect there to be a password in the profile that isn't then stored in the credential store? 

A scenario where that might be the case is if someone put a password field for an integrated auth profile. Not sure what would happen but I would expect we wouldn't necessarily store anything in the credential store. But in that case wouldn't it still be correct to strip out the password from the profile anyways? I'm not sure myself - it's a weird edge case. But it's something to consider since in general I think we should just avoid ever having passwords stored in that file (even if users enter it in themselves)

Reply:
Yes, it was fixing this case - https://github.com/microsoft/vscode-mssql/issues/1438

You manually add a connection in the settings file with save password and it makes a connection. In that case, we need to remove the password from the `json`. This part is the error handling for that feature.